src/main/resources/META-INF/rewrite/examples.yml

```diff
@@ -1212,6 +1213,61 @@ examples:
     language: java
 ---
 type: specs.openrewrite.org/v1beta/example
+recipeName: org.openrewrite.staticanalysis.InstanceOfPatternMatch
+examples:
+- description: ''
+  sources:
+  - before: |
+      class LeftNode {
+          int bar() {
+              return 0;
+          }
+      }
+      class RightNode {
+          int bar() {
+              return 1;
+          }
+      }
+
+      class Foo {
+          void bar(Object o1, Object o2) {
+              if (o1 instanceof LeftNode && o2 instanceof RightNode) {
+                ((LeftNode)o1).bar();
+                ((RightNode)o2).bar();
+              }
+              else if (o1 instanceof RightNode && o2 instanceof LeftNode) {
+                ((RightNode)o1).bar();
+                ((LeftNode)o2).bar();
+              }
+          }
+      }
+    after: |
+      class LeftNode {
+          int bar() {
+              return 0;
+          }
+      }
+      class RightNode {
+          int bar() {
+              return 1;
+          }
+      }
+
+      class Foo {
+          void bar(Object o1, Object o2) {
+              if (o1 instanceof LeftNode node2 && o2 instanceof RightNode node3) {
```

Comment:
Surprising to see node2 and node3 before node and node1.

Reply:
Yes that slightly annoyed me as well, but figured since we're already a few levels deep into edge cases (classes with the same suffix, used in binaries, with an else of overlapping instanceof expressions) I didn't want to explore how to switch those around and risk potentially breaking other logic that had assumed a particular processing order. From a brief look into it it's not as easy as moving down a super.visit call unfortunately.